cmd/hubagent/workload/setup.go

```diff
@@ -384,7 +388,17 @@ func SetupControllers(ctx context.Context, wg *sync.WaitGroup, mgr ctrl.Manager,
 
 		// Set up the scheduler
 		klog.Info("Setting up scheduler")
-		defaultProfile := profile.NewDefaultProfile()
+		profileOpts := profile.ProfileOptions{}
+		if opts.EnableComputeService {
+			klog.Info("Azure capacity service is enabled for scheduler")
+			computeClient, err := compute.NewAttributeBasedVMSizeRecommenderClient(opts.ComputeServiceEndpoint, httputil.DefaultClientForAzure)
+			if err != nil {
+				klog.ErrorS(err, "unable to create Azure AttributeBasedVMSizeRecommenderClient")
+				return err
+			}
+			profileOpts.ComputeService = azure.NewAzureCapacityService(*computeClient)
```

Comment:
Some is called "ComputeService", some is called "CapacityService". need some alignment here cc @zhiying-lin